test/FunctionsV2/CorrelationEndToEndTests.cs

```diff
@@ -135,6 +192,10 @@ public async Task AllOrchestrationActivityActions(bool extendedSessions, string
             options.Tracing = traceOptions;
             var sendAction = new Action<ITelemetry>(
                 delegate(ITelemetry telemetry) { sendItems.Enqueue(telemetry); });
+
+            string currSiteName = Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME");
```

Comment:
We generally use try-catch-finally loops to safely set environment variables if the test fails.

In general, I would like to move away from using environment variable based tests at all. The extension should use the INameResolver injected into the extension constructor to read environment variables, and then we can just mock that in our tests.

Reply:
I can change the test to mock `INameResolver`. Is [this test](https://github.com/Azure/azure-functions-durable-extension/blob/47247f9ab322efdf9cd9216582bc5a7dce9bb37f/test/Common/DurableTaskLifeCycleNotificationTest.cs#L39), specifically [line 46](https://github.com/Azure/azure-functions-durable-extension/blob/47247f9ab322efdf9cd9216582bc5a7dce9bb37f/test/Common/DurableTaskLifeCycleNotificationTest.cs#L46), a good example of how to mock `INameResolver` and inject it into the extension?

Reply:
@bachuv, in general, that is a good way to mock the INameResolver and inject it. As a note, we may need to make sure that our source code is actually reading these environment variables via the `INameResolver` service instead of `Environment.GetEnvironmentVariable`, otherwise this mock approach won't work.